oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie2.py

```diff
@@ -0,0 +1,280 @@
+import os
+import re
+import time
+from pathlib import Path
+from subprocess import Popen
+
+import pandas as pd
+from joblib import Parallel, delayed, parallel_backend
+
+from ._filter_base import ProbeFilterBase
+from ._utils import _load_probes_info, _write_output, _write_removed_genes
+
+
+class ProbeFilterBowtie2(ProbeFilterBase):
+    def __init__(
+        self,
+        n_jobs,
+        dir_output,
+        dir_annotations,
+        file_transcriptome_fasta,
+        min_score=None,
+    ):
+        """This class filters probes based on the Bowtie 2 read alignment tool. It is recommended to use Bowtie 2 instead of Bowtie for reads longer than about 50 bp, as it gives better performance.
+        The user can customize the filtering by specifying the min_score. The Bowtie 2 filter gives a alignmnet score to each probe. The higher the score, the more simlar the read sequence is to the reference sequence. The min_score parameter filters out probes with an aligment score greater than min_score
+
+        Use conda install -c bioconda bowtie2 to install the Bowtie 2 package
+
+        :param file_transcriptome_fasta: path to fasta file containing all probes
+        :type file_transcriptome_fasta: str
+        :param min_score: User defined threshhold for alignmnent score. If specified, the Bowtie 2 filter filters out probes with an aligment score greater than min_score. If None, min_score defaults to -0.6 + -0.6 * L, where L is the read length
+        """
+
+        super().__init__(n_jobs, dir_output, dir_annotations)
+
+        self.n_jobs = n_jobs
+        self.file_transcriptome_fasta = file_transcriptome_fasta
+        self.min_score = min_score
+
+        self.dir_bowtie2 = os.path.join(self.dir_output, "bowtie2")
```

Comment:
we should only write files if necessary (e.g. temp files) and then remove them. Everything else should be stored in probes DB